seshdash/utils/alert.py

```diff
@@ -13,46 +15,81 @@
-def alert_check(data_point):
+# A Sesh_Alert object is created for each 'alert triggered and an email is send if the rule has send_mail option true
+def alert_check(site):
     ops = {'lt': lambda x,y: x<y,
            'gt': lambda x,y: x>y,
            'eq' : lambda x,y: x==y,
     }
-    recipients = []
+    email_recipients = []
+    sms_recipients = []
     content = {}
 
-    rules = Alert_Rule.objects.filter(site = data_point.site)
+    rules = Alert_Rule.objects.filter(site = site)
     for rule in rules:
-        real_value = getattr(data_point,rule.check_field)
+
+        if '#' in rule.check_field: # If the rule is valid (contains model and field_name)
+            model, field_name = rule.check_field.split('#')
+             
+            # Getting the model name and the latest value of the model field
+            model = get_model_from_string(model)
+            data_point = model.objects.all().order_by('-id')[0]
+            real_value = getattr(data_point, field_name)
+
         if ops[rule.operator](real_value,rule.value):
-            content_str = "site:%s\nrule:%s '%s' %s --> found %s " %(data_point.site.site_name,rule.check_field,rule.operator,rule.value,real_value)
+            content_str = "site:%s\nrule:%s '%s' %s --> found %s " %(site.site_name,rule.check_field,rule.operator,rule.value,real_value)
 
             # Get ready content for email
-            content['site'] = data_point.site.site_name
+            content['site'] = site.site_name
             content['alert'] = content_str
             content['time'] = data_point.time
             content['data_point'] = data_point
 
             # TODO rule object should have the list of related persons to send alert mail
-            users = get_users_with_perms(data_point.site)
+            users = get_users_with_perms(site)
+            # TODO to be removed
+            userSms = Sesh_User.objects.all()
             for user in users:
-                recipients.append(user.email)
-            logging.debug("emailing %s" %recipients)
+                email_recipients.append(user.email)
+            
+            for user in userSms:
+                if user.phone_number and user.on_call:
+                    sms_recipients.append(user.phone_number)
+
+            logging.debug("emailing %s" %email_recipients)
             #print "emailing %s"%recipients
             # recipients = ["seshdash@gmail.com",]
             alert_obj = Sesh_Alert.objects.create(
-                    site = data_point.site,
+                    site = site,
                     alert=rule, date=timezone.now(),
                     isSilence=False,
-                    alertSent=rule.send_mail,
-                    point=data_point)
+                    emailSent=False,
+                    slackSent=False,
+                    smsSent=False,
+                    point_model=type(data_point).__name__ )
+            alert_obj.save()
+            
+            print "The alert points to: ",
```

Comment:
please remove print statments and replace with propper logger.debug or logger.info statements